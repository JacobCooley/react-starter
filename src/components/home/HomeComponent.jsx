import React from 'react';
import RedditContainer from './RedditContainer';

const HomeComponent = ({ count, onIncrementClick, onDecrementClick }) => (
	<div>
		<p>current count: {count}</p>
		<button onClick={onIncrementClick}>Increment</button>
		<button onClick={onDecrementClick}>Decrement</button>
		<RedditContainer/>
	</div>
)

export default HomeComponent;