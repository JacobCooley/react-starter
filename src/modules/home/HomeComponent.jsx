import React from 'react';
import RedditContainer from './RedditContainer';
import Button from '@components/Button'
import PropTypes from "prop-types";

const HomeComponent = ({ count, onIncrementClick, onDecrementClick }) => (
	<div>
		<p>current count: {count}</p>
		<Button onClick={onIncrementClick} text='Increment' />
		<Button onClick={onDecrementClick} text='Decrement' />
		<RedditContainer/>
	</div>
)

HomeComponent.propTypes = {
	count: PropTypes.number,
	onIncrementClick: PropTypes.func,
	onDecrementClick: PropTypes.func,
};

HomeComponent.defaultProps = {
	count: 0,
	onIncrementClick: () => {},
	onDecrementClick: () => {},
};

export default HomeComponent;