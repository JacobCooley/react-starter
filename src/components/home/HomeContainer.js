import { connect } from 'react-redux';
import HomeComponent from './HomeComponent';
import { homeOperations } from './duck';

const mapStateToProps = state => {
    const { count } = state.home;
    return { count }
};

const mapDispatchToProps = dispatch => {
    const onIncrementClick = () => dispatch(homeOperations.increment(1));
    const onDecrementClick = () => dispatch(homeOperations.decrement(1));

    return {
        onIncrementClick,
        onDecrementClick
    }
};

const HomeContainer = connect(
    mapStateToProps,
    mapDispatchToProps
)(HomeComponent);

export default HomeContainer;