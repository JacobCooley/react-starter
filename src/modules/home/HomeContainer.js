import { connect } from 'react-redux';
import HomeComponent from './HomeComponent';
import { homeOperations } from './duck';
import PropTypes from 'prop-types'

const mapStateToProps = state => {
    const { count } = state.home;
    return { count }
};

const mapDispatchToProps = dispatch => {
    const onIncrementClick = () => dispatch(homeOperations.increment(1));
    const onDecrementClick = () => dispatch(homeOperations.decrement(1));

    return {
        onIncrementClick,
        onDecrementClick
    }
};

HomeComponent.propTypes = {
	count: PropTypes.number,
	onIncrementClick: PropTypes.func,
	onDecrementClick: PropTypes.func,
};

HomeComponent.defaultProps = {
	count: 0,
	onIncrementClick: () => {},
	onDecrementClick: () => {},
};

const HomeContainer = connect(
    mapStateToProps,
    mapDispatchToProps
)(HomeComponent);

export default HomeContainer;