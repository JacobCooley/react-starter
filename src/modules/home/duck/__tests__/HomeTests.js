import HomeComponent from '@home/HomeComponent'
import HomeContainer from '@home/HomeContainer'
import React from 'react'
import jest from 'jest'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'

configure({ adapter: new Adapter() })

describe('<HomeComponent /> --- Snapshot',()=>{
	it('+++ capturing Snapshot of <HomeComponent />', () => {
		const renderedValue =  shallow(<HomeComponent count={0}/>)
		expect(renderedValue).toMatchSnapshot();
	});
	
	// it('+++ verify PropTypes', () => {
	// 	const props = {
	// 			count: 0,
	// 			onIncrementClick: jest.fn(),
	// 			onDecrementClick: jest.fn()
	// 		},
	// 	DateInputComponent = mount(<HomeComponent {...props} />).find('.datepicker');
	// 	expect(DateInputComponent.hasClass('react-datepicker-hide-month')).toEqual(true);
	// });
});

describe('<HomeComponent /> shallow copy', () => {
	let wrapper
	const count = 0
	
	beforeEach(() => {
		wrapper = shallow(<HomeComponent count={count}/>)
		
	})
	
	it('+++ render the DUMB component', () => {
		expect(wrapper.length).toEqual(1)
	})
	
	it('+++ contains count', () => {
		expect(wrapper.find('p').text()).toEqual('current count: ' + count)
	})
})

describe('<HomeContainer /> shallow copy', () => {
	const initialState = {
		home: {
			count: 0
		}
	}
	const mockStore = configureStore()
	let store, container
	
	beforeEach(() => {
		store = mockStore(initialState)
		container = shallow(<HomeContainer store={store}/>)
	})
	
	it('+++ render the connected(SMART) component', () => {
		expect(container.length).toEqual(1)
	})
	
	it('+++ check Prop matches with initialState', () => {
		expect(container.prop('count')).toEqual(initialState.home.count)
	})
	
})
