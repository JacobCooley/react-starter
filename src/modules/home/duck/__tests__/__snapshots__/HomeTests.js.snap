// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeComponent /> --- Snapshot +++ capturing Snapshot of <HomeComponent /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeComponent
    count={0}
    onDecrementClick={[Function]}
    onIncrementClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          current count: 
          0
        </p>,
        <Button
          onClick={[Function]}
          text="Increment"
        />,
        <Button
          onClick={[Function]}
          text="Decrement"
        />,
        <Connect(RedditComponent) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "current count: ",
            0,
          ],
        },
        "ref": null,
        "rendered": Array [
          "current count: ",
          0,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "text": "Increment",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "text": "Decrement",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            current count: 
            0
          </p>,
          <Button
            onClick={[Function]}
            text="Increment"
          />,
          <Button
            onClick={[Function]}
            text="Decrement"
          />,
          <Connect(RedditComponent) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "current count: ",
              0,
            ],
          },
          "ref": null,
          "rendered": Array [
            "current count: ",
            0,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "text": "Increment",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "text": "Decrement",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
